### PR TITLE
Add some inline comments to the new parser impl

### PR DESCRIPTION
**Pull request by KodrAus**
cc @Nugine @QnnOkabayashi 

This just adds some inline comments to the parser implementation so it's easier for new contributors to follow.

### CODE REVIEW

**Review by QnnOkabayashi — APPROVED** [review]
Darn I would've liked to have reviewed Nugine's PR but I didn't see it before this. Looks good though!

File: shared/parser.rs [hunk 1]
```diff
@@ -31,47 +38,73 @@ pub const fn try_parse(input: &str) -> Result<[u8; 16], InvalidUuid> {
 
 #[inline]
 const fn parse_simple(s: &[u8]) -> Result<[u8; 16], ()> {
+    // Should be optimized away as redundant
     if s.len() != 32 {
         return Err(());
     }
```
Comment (Nugine) [inline shared/parser.rs:44]: The length check removes all the other bound checks in this function.

File: shared/parser.rs [hunk 2]
```diff
@@ -31,47 +38,73 @@ pub const fn try_parse(input: &str) -> Result<[u8; 16], InvalidUuid> {
...
+
     Ok(buf)
 }
 
 #[inline]
 const fn parse_hyphenated(s: &[u8]) -> Result<[u8; 16], ()> {
+    // Should be optimized away as redundant
     if s.len() != 36 {
         return Err(());
     }
```
Comment (Nugine) [inline shared/parser.rs:75]: Same as above

**Review by Nugine — APPROVED** [review]
LGTM

**Comment by KodrAus** [comment]
Sorry @QnnOkabayashi! I'll remember to keep you CC'd on these 🙇

**Pull request merged**